### PR TITLE
Changes for Mixpanel stats

### PR DESCRIPTION
**Pull request by samartnik**
Changes required for Mixpanel statistics

### CODE REVIEW

**Comment by avently** [comment]
Wow. It is so cool that now you are tracking your users. I love that. Waited so long when browser that should care about a user just ruins his privacy. Most have feature. It's now like Chrome without Google but with Brave

File: chrome/android/java/src/org/chromium/chrome/browser/ChromeActivity.java [hunk 1]
```diff
@@ -1385,12 +1385,14 @@ public void finishNativeInitialization() {
...
+            if (!ConfigAPIs.MIXPANEL_TOKEN.isEmpty()) {
                 app.mMixpanelInstance = MixpanelAPI.getInstance(getApplicationContext(), ConfigAPIs.MIXPANEL_TOKEN);
             } else {
                 Log.i("ChromeActivity", "MixPanel is not activated");
-            }*/
-            //MixPanelWorker.SendEvent("MainActivity - onCreate called", "start", "open");
+            }
+            MixPanelWorker.SendBraveAppStartEvent(PrefServiceBridge.getInstance().isHTTPSEEnabled(), PrefServiceBridge.getInstance().isTrackingProtectionEnabled(),
+            PrefServiceBridge.getInstance().isAdBlockEnabled(), PrefServiceBridge.getInstance().isAdBlockRegionalEnabled(),
+            PrefServiceBridge.getInstance().isFingerprintingProtectionEnabled());
```
Comment (SergeyZhukovsky) [inline chrome/android/java/src/org/chromium/chrome/browser/ChromeActivity.java]: Probably it is nice to have an info about JavaScript enabled or not

**Comment by diracdeltas** [comment]
@avently you are correct, this PR is currently unacceptable. 

Proposed modifications:
* Add option to disable all telemetry
* Modify the mixpanel library to strip out all fields except the randomly-generated GUID (rotate every month to limit tracking persistence) and the app version by default.

With the above changes, we would be sending 5 bits of user-identifiable data in this PR (the brave shield settings).

Note that we don't connect to any third party tracking servers directly, only to metric-proxy.brave.com (source code: https://github.com/brave/metric-proxy) which at least prevents mixpanel's API server from getting your IP address.

**Comment by samartnik** [comment]
Made changes according to proposed modifications.

**Comment by SergeyZhukovsky** [comment]
++
@diracdeltas please review it as well

File: chrome/android/java/src/org/chromium/chrome/browser/MixPanelWorker.java [hunk 2]
```diff
@@ -1,17 +1,62 @@
...
+
+    // Send event with no options
+    public static void SendEvent(String eventName) {
+        if (!ContextUtils.getAppSharedPreferences().getBoolean(PREF_SEND_METRICS, true)) {
+            return;
+        }
         ChromeApplication app = (ChromeApplication)ContextUtils.getApplicationContext();
+        GetMixpanelInstance(app);
+        if (null != app && null != app.mMixpanelInstance) {
+            app.mMixpanelInstance.track(eventName);
```
Comment (diracdeltas) [inline chrome/android/java/src/org/chromium/chrome/browser/MixPanelWorker.java]: Should this be modified to strip out most of the mixpanel default tracking data? (see https://github.com/brave/internal/issues/80#issuecomment-325834552 list item 3)

Sorry if this is done elsewhere and I just missed it

Comment (samartnik) [reply chrome/android/java/src/org/chromium/chrome/browser/MixPanelWorker.java]: It is done inside Mixpanel library 
third_party/mixpanel/lib/mixpanel.jar

**Pull request merged**